tests/unittests/objects/jira/test_jira.py

```diff
@@ -1,11 +1,52 @@
+import pytest
 from jira import Issue
 
 
-def test_get_issue_by_id_returns_jira_issue_from_jira_api_client_with_matching_issue_id(
-    patch_jira_api_requests,
-    jira,
-    fake_issue_id,
-):
+@pytest.fixture
+def jira(patch_jira_api_requests, jira):
+    jira._caps = patch_jira_api_requests
+    yield jira
+
+
+@pytest.fixture
+def fake_issue(fake_issue_id, jira) -> Issue:
+    yield jira.get_issue_by_id_or_key(issue=fake_issue_id)
+
+
+@pytest.fixture
+def fake_attachment_path(tmp_path):
+    path = tmp_path.joinpath("build_log.json")
+    path.write_text("anything")
+    yield path
+
+
+def test_get_issue_by_id_returns_jira_issue_from_jira_api_client_with_matching_issue_id(fake_issue_id, jira):
     issue = jira.get_issue_by_id_or_key(issue=fake_issue_id)
     assert isinstance(issue, Issue)
     assert issue.id == fake_issue_id
+
+
+def test_add_file_attachment_with_success(fake_issue, jira, fake_attachment_path):
+    jira.add_attachment_to_issue(
+        issue=fake_issue,
+        attachment_path=fake_attachment_path.as_posix(),
+    )
+
+
+class TestWithPermissionFailures:
```

Comment:
Let's just add the fixture to the top of the file with the others and remove the class? Unless for some reason this needs to be here?